### PR TITLE
Benchmark versus dynamic grid and mentioning

### PR DESCRIPTION
**Pull request by Datseris**
C.f. https://github.com/cesaraustralia/DynamicGrids.jl/issues/10

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -76,8 +76,23 @@ The package is in Julia's package list. Install it using this command:
...
+Agents.jl targets complicated agent based models (ABMs) that are defined on truly arbitrary complex graphs.
+Because of this, the core datastructure of this package is a dictionary that maps unique IDs to Agents, as shown in the [Tutorial](@ref) and specifically in [`AgentBasedModel`](@ref).
+As all agents are unique entities, if one "dies" it is entirely and forever removed from memory.
+Similarly, when a new agent becomes "alive", this literally means that a new agent datastructure is initialized and added to this dictionary.
+
+This design decision allows for arbitrary ABMs and also allows flexible implementation on arbitrary graphs, as well as makes the data collection functionality of Agents.jl powerful and intuitive.
+But this is not necessary for *cellular-automata-like* models, where the grid cell and the "agent" identity are fully equivalent.
+This is the case e.g. in the [Forest fire](@ref) and [Game of life](@ref) examples.
+For such applications the Julia package [DynamicGrids.jl](@ref) can give up to 2 orders of magnitude performance gain than Agents.jl (see `/test/benchmark/DynamicGrids`), while also allowing advanced visualization.
```
Comment (kavir1698) [inline docs/src/index.md]: Rewording: can be up to 2 orders of magnitude faster than Agents.jl

Comment (Datseris) [reply docs/src/index.md]: I'll modify this a bit in general.

**Comment by Datseris** [comment]
I've reworked the entire homepage: I've noticed that there was a lot of repetetion happening, most things were being mentioned twice. I've reduced the amount of text almost by half.

I've also noticed that the comparison with DynamicGrids.jl was not entirely fair, as Agents.jl has other advantages, like e.g. stability and simpler API. I've also realized that the amount of cases you would scientifically prefer to use Agents.jl is actually bigger than what I initially thought (see bullet points).

@kavir1698 feel free to merge this once you have a glance.

File: examples/forest_fire.jl [hunk 2]
```diff
@@ -15,6 +15,11 @@
 # This model is an example that does _not_ have an `agent_step!` function. It only
 # uses a `model_step!`
 
+# !!! info "Cellular-automata-like ABMs"
+#     The forest fire model is a cellular automaton.
+#     Have a read at [Alterantive package: DynamicGrids.jl](@ref), which can do
```
Comment (Datseris) [inline examples/forest_fire.jl]: ah this is incorrect now

**Pull request merged**